features/steps/ubuntu_advantage_tools.py

```diff
@@ -1,37 +1,161 @@
 import logging
+import os
 import re
 
 from behave import when
 
-from features.environment import UA_PPA_TEMPLATE, build_debs_from_sbuild
-from features.steps.machines import launch_machine
-from features.steps.shell import (
-    when_i_run_command,
-    when_i_run_command_on_machine,
-    when_i_run_shell_command,
-)
-from features.util import InstallationSource
+from features.steps.files import when_i_create_file_with_content
+from features.steps.packages import when_i_apt_install
+from features.steps.shell import when_i_run_command, when_i_run_shell_command
+from features.util import SUT, InstallationSource, build_debs
+
+
+@when("I install ubuntu-advantage-tools")
+def when_i_install_uat(context, machine_name=SUT):
+    instance = context.machines[machine_name].instance
+    series = context.machines[machine_name].series
+    is_pro = "pro" in context.config.machine_type
+    if context.config.install_from is InstallationSource.ARCHIVE:
+        instance.execute("sudo apt update")
+        when_i_apt_install(
+            context, "ubuntu-advantage-tools", machine_name=machine_name
+        )
+        if is_pro:
+            when_i_apt_install(
+                context, "ubuntu-advantage-pro", machine_name=machine_name
+            )
+    elif context.config.install_from is InstallationSource.PREBUILT:
+        debs_path = context.config.debs_path
+        deb_paths = [
+            os.path.join(debs_path, deb_file)
+            for deb_file in os.listdir(debs_path)
+            if series in deb_file
+        ]
+        logging.info("using debs: {}".format(deb_paths))
+        for deb_path in deb_paths:
+            if "pro" not in deb_path or is_pro:
+                instance.push_file(deb_path, "/tmp/behave_ua.deb")
+                when_i_apt_install(
+                    context, "/tmp/behave_ua.deb", machine_name=machine_name
+                )
+                instance.execute("sudo rm /tmp/behave_ua.deb")
+    elif context.config.install_from is InstallationSource.LOCAL:
+        deb_paths = build_debs(series)
+        for deb_path in deb_paths:
```

Comment:
Since we know the order we will receive the packages from `build_debs`, I think we can change this code to:
```python
ua_deb_path, pro_deb_path = build_debs(series)
for deb_path in (ua_deb_path, pro_deb_path):
    instance.push_file(deb_path, "/tmp/behave_ua.deb")
    when_i_apt_install(
        context, "/tmp/behave_ua.deb", machine_name=machine_name
     )
     instance.execute("sudo rm /tmp/behave_ua.deb")
```